src/rshim.c

```diff
@@ -2142,11 +2142,11 @@ int rshim_set_drop_mode(rshim_backend_t *bd, int value)
 static int rshim_check_locked_mode(rshim_backend_t *bd)
 {
     int rc;
-    uint64_t value;
+    uint64_t value = 0;
 
     rc = bd->read_rshim(bd, RSHIM_CHANNEL, bd->regs->scratchpad1, &value,
                         RSHIM_REG_SIZE_8B);
-    if (rc < 0) {
+    if (rc < 0 || RSHIM_BAD_CTRL_REG(value)) {
```

Comment:
It's not clear why "RSHIM_BAD_CTRL_REG" is specifically added here but not in other "bd->read_rshim()" calls. If there's a specific reason we only need it here in "rshim_check_locked_mode()", maybe explain it here. Or maybe move this check inside rshim_pcie_read()?

Reply:
RSHIM_BAD_CTRL_REG is used in several other places as well, which could potentially happen in certain corner cases such as during ARM reset. It's mainly used for registers that we know the content. It's not reliable to be used for generic register read.